src/pages/ConnectPayment/components/AdyenCheckout.tsx

```diff
@@ -0,0 +1,207 @@
+import { useHistory } from 'react-router'
+import { SpinnerWrapper } from './Spinner'
+
+const Wrapper = styled('div')`
+  position: relative;
+
+  @media (min-width: 801px) {
+    padding-right: 5rem;
+  }
+
+  .adyen-checkout__payment-method {
+    &,
+    &:first-child,
+    &:last-child {
+      border-radius: 3px;
+      border-top-left-radius: 3px;
+      border-top-right-radius: 3px;
+    }
+  }
+  .adyen-checkout__payment-method--selected {
+    background: ${colorsV2.offwhite};
+  }
+`
+
+export const AdyenCheckout = () => {
+  const availablePaymentMethods = useAvailablePaymentMethodsQuery()
+  const adyenCheckoutRef = React.useRef<HTMLDivElement>()
+  const [adyenLoaded, setAdyenLoaded] = React.useState(false)
+  const [tokenizePaymentMutation] = useTokenizePaymentDetailsMutation()
+  const history = useHistory()
+  const currentLocale = useCurrentLocale()
+
+  const paymentMethodsResponse =
+    availablePaymentMethods.data?.availablePaymentMethods.paymentMethodsResponse
+
+  React.useEffect(() => {
+    if (!paymentMethodsResponse || !adyenLoaded) {
+      return
+    }
+
+    createAdyenCheckout({
+      currentLocale,
+      paymentMethodsResponse,
+      tokenizePaymentMutation,
+      history,
+    }).mount(adyenCheckoutRef.current)
+  }, [paymentMethodsResponse, adyenLoaded])
+
+  React.useEffect(mountAdyenJs(setAdyenLoaded), [])
+
+  React.useEffect(mountAdyenCss, [])
+
+  return (
+    <Wrapper>
+      {!adyenLoaded && (
+        <SpinnerWrapper>
+          <Spinner />
+        </SpinnerWrapper>
+      )}
+      <div
+        ref={adyenCheckoutRef as React.MutableRefObject<HTMLDivElement>}
+      ></div>
+    </Wrapper>
+  )
+}
+
+interface AdyenCheckoutProps {
+  currentLocale: string
+  paymentMethodsResponse: Scalars['PaymentMethodsResponse']
+  tokenizePaymentMutation: TokenizePaymentDetailsMutationFn
+  history: ReturnType<typeof useHistory>
+}
+
+const createAdyenCheckout = ({
+  currentLocale,
+  paymentMethodsResponse,
+  tokenizePaymentMutation,
+  history,
+}: AdyenCheckoutProps) => {
+  const locale = match([
+    ['sv_SE', 'sv-SE'],
+    ['en_SE', 'en-US'],
+    ['nb_NO', 'no-NO'],
+    ['en_NO', 'en-US'],
+  ])(getLocaleIsoCode(currentLocale))
+
+  const returnUrl = `${window.location.origin}${
+    currentLocale ? '/' + currentLocale : ''
+  }/new-member/connect-payment` // FIXME maybe this should be /download?
+
+  const configuration = {
+    locale,
+    environment: 'test',
+    originKey: (window as any).ADYEN_ORIGIN_KEY,
+    paymentMethodsResponse: JSON.parse(paymentMethodsResponse),
+    paymentMethodsConfiguration: {
+      card: {
+        hasHolderName: true,
+        holderNameRequired: true,
+      },
+    },
+    enableStoreDetails: true,
+    returnUrl,
+    // onChange: console.log,
+    onAdditionalDetails: (state: any, _dropinComponent: any) => {
+      // TODO call to additional details mutation?
+      // tslint:disable-next-line no-console
+      console.log(state)
+    },
```

Comment:
In case you want to clean up

Reply:
I don't haha :D this is needed for 3d secure but not otherwise afak. So i'd like to leave it just to not have to look up again how to do things.